sled-agent/src/rack_setup/plan/service.rs

```diff
@@ -119,6 +124,54 @@ pub struct SledConfig {
     pub zones: Vec<BlueprintZoneConfig>,
 }
 
+impl SledConfig {
+    /// Adds a zone to the Sled's configuration, as well as any number of
+    /// durable datasets.
+    pub fn add_zone_and_datasets(&mut self, zone: BlueprintZoneConfig) {
```

Comment:
This is RSS's version of "whenever you add a zone, make sure you track all possible datasets used by that zone too".